src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -81,74 +94,114 @@ void KevinHallModel::update_risk_factors(RuntimeContext &context) {
         if (person.age == 0) {
             initialise_nutrient_intakes(person);
             initialise_energy_intake(person);
-            initialise_weight(person);
         } else {
             update_nutrient_intakes(person);
             update_energy_intake(person);
         }
     }
 
-    // TODO: Compute target body weight.
-    const float target_BW = 100.0;
-
-    // Trial run.
-    double mean_sim_body_weight = 0.0;
-    double mean_adjustment_coefficient = 0.0;
+    // Update: different model for children and adults.
     for (auto &person : context.population()) {
         // Ignore if inactive.
         if (!person.is_active()) {
             continue;
         }
 
-        // Simulate person and compute adjustment coefficient.
-        SimulatePersonState state = simulate_person(person, 0.0);
-        mean_sim_body_weight += state.BW;
-        mean_adjustment_coefficient += state.adjust;
+        if (person.age < kevin_hall_age_min) {
+            initialise_weight(person);
+        } else {
+            kevin_hall_run(person);
+        }
+    }
+
+    KevinHallAdjustmentTable adjustments;
+
+    // Baseline scenatio: compute adjustments.
+    if (context.scenario().type() == ScenarioType::baseline) {
+        adjustments = compute_kevin_hall_adjustments(context.population());
     }
 
-    // Compute model adjustment term.
-    const size_t population_size = context.population().current_active_size();
-    mean_sim_body_weight /= population_size;
-    mean_adjustment_coefficient /= population_size;
-    double shift = (target_BW - mean_sim_body_weight) / mean_adjustment_coefficient;
+    // Intervention scenario: recieve adjustments from baseline scenario.
+    else {
+        auto message = context.scenario().channel().try_receive(context.sync_timeout_millis());
+        if (!message.has_value()) {
+            throw core::HgpsException(
+                "Simulation out of sync, receive Kevin Hall adjustments message has timed out");
+        }
+
+        auto &basePtr = message.value();
+        auto *messagePrt = dynamic_cast<KevinHallAdjustmentMessage *>(basePtr.get());
+        if (!messagePrt) {
+            throw core::HgpsException(
+                "Simulation out of sync, failed to receive a Kevin Hall adjustments message");
+        }
 
-    // Final run.
+        adjustments = messagePrt->data();
+    }
```

Comment:
I'd make most of this block into a little helper function, returning a `std::optional<KevinHallAdjustmentMessage>`, then you can write:

```cpp
    // Intervention scenario: recieve adjustments from baseline scenario.
    else if (const auto &message = try_get_adjustment_message(context))
        adjustments = message->data();
    else
        throw core::HgpsException(
                "Simulation out of sync, failed to receive a Kevin Hall adjustments message");
```